src/dsync/dsync.c

```diff
@@ -963,8 +963,18 @@ static int dsync_strmap_compare_data(
         /* ignore prefix portion of path to use as key */
         name += strlen_prefix;
 
-        /* get comparison results for this item */
-        int flag = results[i];
+        mfu_filetype type = mfu_flist_file_get_type(src_compare_list, i);
+        int flag = 0;
+        if (type == MFU_TYPE_LINK) {
```

Comment:
At this point, we have already compared file contents.  If we find the symlinks point to different targets, didn't we compare the wrong contents, and the results of the file comparison results aren't valid?

It seems like we need to do this earlier.

Reply:
IF I understand the dsync_strmap_compare_data function, we build a list of regular file with the chunk size to compare. The symlink are not in the list, only the files with chuck get compared. The first compare loop cannot be used without serious modification.  But I also agree the symlink compare is done in a loop used for a different purpose. It should be done before.  I suggest another loop after the loop for regular files with chunk size.

Reply:
For dsync, I moved the symlink destination comparison at the upper level in `dsync_strmap_compare()`, similarly to `dcmp_strmap_compare()`. This avoids messing up with the logic in `dsync_strmap_compare_data()`  and `dsync_strmap_compare_lite()` that are specifically designed for regular files. Tell me what you think!